### ISSUE TITLE
make install with modules not correctly passing state of modules

### ISSUE THREAD

**Issue by cpaxton** [cmake-basis/legacy#548]
I installed a project with modules for the first time and the config script doesn't seem to be working correctly when it is loaded from outside.

after running:

```
mkdir build
cd build
cmake ..
ccmake . # turn on relevant modules
make
make install
```

from `ccmake .`

```
 BUILD_ALL_MODULES                ON       
               ...snip...                                                                                
 MODULE_FRI-Client-SDK_Cpp        ON                                                                                                                                             
 MODULE_grl                       ON                                                                                                                                             
 MODULE_robone                    ON                                                                                                                                             
 MODULE_roboneprivate             ON        
```

In a third outside package I go to load the installed library

with this cmake line:

```
find_package(robonetracker REQUIRED COMPONENT grl)
```

I run with the following (which passes the -D parameter to a normal cmake run from an extrnal wrapper, so I believe it is a detail which can be overlooked):

```
catkin build --this --cmake-args -Drobonetracker_DIR=/opt/robonetracker/lib/cmake/robonetracker/
```

when I run I get the following error:
```
CMake Error at /opt/robonetracker/lib/cmake/robonetracker/robonetrackerConfig.cmake:165 (message):
  Requested module COMPONENT of package robonetracker was not enabled when
  building robonetracker.  Rebuild the robonetracker package with
  MODULE_COMPONENT set to ON and try again.
Call Stack (most recent call first):
  CMakeLists.txt:14 (find_package)
```

@ahundt is also involved in this

**Comment by cpaxton** [comment]
Also tested with:

```
cmake .. -Drobonetracker_DIR=/opt/robonetracker/lib/cmake/robonetracker/
```

Results:
```
-- The C compiler identification is GNU 4.8.4
-- The CXX compiler identification is GNU 4.8.4
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Using CATKIN_DEVEL_PREFIX: /home/cpaxton/catkin_ws/src/grl_ros_driver/build/devel
-- Using CMAKE_PREFIX_PATH: /home/cpaxton/catkin_ws/devel;/opt/ros/indigo
-- This workspace overlays: /home/cpaxton/catkin_ws/devel;/opt/ros/indigo
-- Found PythonInterp: /usr/bin/python (found version "2.7.6") 
-- Using PYTHON_EXECUTABLE: /usr/bin/python
-- Using Debian Python package layout
-- Using empy: /usr/bin/empy
-- Using CATKIN_ENABLE_TESTING: ON
-- Call enable_testing()
-- Using CATKIN_TEST_RESULTS_DIR: /home/cpaxton/catkin_ws/src/grl_ros_driver/build/test_results
-- Looking for include file pthread.h
-- Looking for include file pthread.h - found
-- Looking for pthread_create
-- Looking for pthread_create - not found
-- Looking for pthread_create in pthreads
-- Looking for pthread_create in pthreads - not found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - found
-- Found Threads: TRUE  
-- Found gtest sources under '/usr/src/gtest': gtests will be built
-- Using Python nosetests: /usr/bin/nosetests-2.7
-- catkin 0.6.16
CMake Error at /opt/robonetracker/lib/cmake/robonetracker/robonetrackerConfig.cmake:165 (message):
  Requested module COMPONENT of package robonetracker was not enabled when
  building robonetracker.  Rebuild the robonetracker package with
  MODULE_COMPONENT set to ON and try again.
Call Stack (most recent call first):
  CMakeLists.txt:14 (find_package)
```

**Comment by schuhschuh** [comment]
Thanks @cpaxton, this is related to issue #537 and some other error in the generated ```<Pkg>Config.cmake``` file. Namely, the ```foreach``` loop variable is not referenced correctly within the loop. This has been fixed now in the "develop" branch.

**Comment by schuhschuh** [comment]
This should be fixed in [version 3.3.0](https://github.com/cmake-basis/BASIS/releases/tag/v3.3.0).
